src/Processors/QueryPlan/ReadFromObjectStorageStep.cpp

```diff
@@ -160,8 +178,8 @@ static bool isPrefixInputOrder(InputOrderInfoPtr small_input_order, InputOrderIn
 
     for (size_t i = 0; i < small_input_order->sort_description_for_merging.size(); ++i)
     {
-        if (!small_input_order->sort_description_for_merging.at(i).column_name.ends_with(
-                big_input_order->sort_description_for_merging.at(i).column_name))
+        if (normalizeColumnName(small_input_order->sort_description_for_merging.at(i).column_name) !=
```

Comment:
Please take a look how is it done in ReadFromMergeTree, I doubt that we need to do such a weird hacks with regexps https://github.com/ClickHouse/ClickHouse/blob/44be6ebba1e3fff60d66b7bda2a55834bde7599e/src/Processors/QueryPlan/ReadFromMergeTree.cpp#L2307-L2337